packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -104,24 +167,330 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+  }, [debouncedQueryUpdate]);
+
+  /* @conditional-compile-remove(mention) */
+  const handleOnChange = useCallback(
+    async (
+      event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>,
+      updatedValue?: string | undefined
+    ): Promise<void> => {
+      let newValue = updatedValue;
+      if (newValue === undefined) {
+        newValue = '';
+      }
+      const triggerText = mentionLookupOptions?.trigger ?? defaultMentionTrigger;
+
+      const newTextLength = newValue.length;
+      let selectionEnd = textFieldRef?.current?.selectionEnd || -1;
+      if (selectionEnd < 0) {
+        selectionEnd = 0;
+      } else if (selectionEnd > newTextLength) {
+        selectionEnd = newTextLength - 1;
+      }
+      // If we are enabled for lookups,
+      if (mentionLookupOptions !== undefined) {
+        // Look at the range of the change for a trigger character
+        const triggerPriorIndex = newValue?.lastIndexOf(triggerText, selectionEnd - 1);
+        // Update the caret position, if not doing a lookup
+        setCaretPosition(Caret.getRelativePosition(event.currentTarget));
+
+        if (triggerPriorIndex !== undefined) {
+          // trigger is found
+          const isSpaceBeforeTrigger = newValue?.substring(triggerPriorIndex - 1, triggerPriorIndex) === ' ';
+          const wordAtSelection = newValue?.substring(triggerPriorIndex, selectionEnd);
+          let tagIndex = currentTriggerStartIndex;
+          if (!isSpaceBeforeTrigger && triggerPriorIndex !== 0) {
+            //no space before the trigger <- continuation of the previous word
+            tagIndex = -1;
+            setCurrentTriggerStartIndex(tagIndex);
+          } else if (wordAtSelection === triggerText) {
+            // start of the mention
+            tagIndex = selectionEnd - triggerText.length;
+            if (tagIndex < 0) {
+              tagIndex = 0;
+            }
+            setCurrentTriggerStartIndex(tagIndex);
+          }
+
+          if (tagIndex === -1) {
+            updateMentionSuggestions([]);
+          } else {
+            // In the middle of a @mention lookup
+            if (tagIndex > -1) {
+              const query = wordAtSelection.substring(triggerText.length, wordAtSelection.length);
+              if (query !== undefined) {
+                await debouncedQueryUpdate(query);
+              }
+            }
+          }
+        }
+      }
+      let result = '';
+      if (tagsValue.length === 0) {
+        // no tags in the string, textValue is a sting
+        result = newValue;
+      } else {
+        // there are tags in the text value, textValue is html string
+        const { changeStart, oldChangeEnd, newChangeEnd } = findStringsDiffIndexes(
+          inputTextValue,
+          newValue,
+          selectionEnd
+        );
+        // get updated html string
+        const change = newValue.substring(changeStart, newChangeEnd);
+        const [updatedHTML, updatedChangeNewEndIndex] = updateHTML(
```

Comment:
So we are parse html to text then parse it back, that's an amazing process

Reply:
We're potentially given HTML that we can only edit as plain text.